### PR TITLE
better snapshot diff + testing-related changes.

### PR DESCRIPTION
**Pull request by evanmcc**
Now that we've merged snapshots in core, we're doing some more testing and finding some issues.  In addition, I've included more debugging facilities.

### CODE REVIEW

File: src/ledger/v1/blockchain_ledger_snapshot_v1.erl [hunk 1]
```diff
@@ -279,27 +281,38 @@ import(Chain, SHA,
...
+                                      ok;
+                                  _ ->
+                                      ok = blockchain:save_block(Block, Chain)
+                              end,
+                              case Ht > Curr2 of
+                                  %% we need some blocks before for history, only absorb if they're
+                                  %% not on the ledger already
+                                  true ->
+                                      lager:info("loading block ~p", [Ht]),
+                                      Rescue = blockchain_block:is_rescue_block(Block),
```
Comment (Vagabond) [inline src/ledger/v1/blockchain_ledger_snapshot_v1.erl:303]: We should have a generic hooks function we can call from all 3 places

File: src/ledger/v1/blockchain_ledger_snapshot_v1.erl [hunk 2]
```diff
@@ -309,6 +322,17 @@ import(Chain, SHA,
...
+get_blocks(Chain) ->
+    Ledger = blockchain:ledger(Chain),
+    DLedger = blockchain_ledger_v1:mode(delayed, Ledger),
+    {ok, Height} = blockchain_ledger_v1:current_height(Ledger),
+    {ok, DHeight} = blockchain_ledger_v1:current_height(DLedger),
+    [begin
+         {ok, B} = blockchain:get_block(N, Chain),
+         B
+     end
+     || N <- lists:seq(max(2, DHeight - 181), Height)].
```
Comment (Vagabond) [inline src/ledger/v1/blockchain_ledger_snapshot_v1.erl:334]: magic number

Comment (evanmcc) [reply src/ledger/v1/blockchain_ledger_snapshot_v1.erl:334]: I'm not even sure it's the right one.  I for local node's I've had to add the genesis block to get things working enough, which will be the case on the spots.   I'm not sure that just storing the genesis block will be enough to get things working right, so we may need to do more work there.  Anyway, I'll fix this for the cases we know about and we can leave the other stuff to another pass.

File: src/ledger/v1/blockchain_ledger_snapshot_v1.erl (hunk 1 above)
Comment (evanmcc) [reply src/ledger/v1/blockchain_ledger_snapshot_v1.erl:303]: yeah, that'd be a good cleanup.  There are too many places where we do almost the same thing.

File: src/ledger/v1/blockchain_ledger_v1.erl [hunk 3]
```diff
@@ -481,7 +481,7 @@ raw_fingerprint(#ledger_v1{mode = Mode} = Ledger, Extended) ->
            subnets = SubnetsCF
           } = SubLedger,
         %% NB: keep in sync with upgrades macro in blockchain.erl
-        Filter = [<<"gateway_v2">>],
+        Filter = [<<"gateway_v2">>, <<"hex_targets">>, <<"gateway_oui">>],
```
Comment (Vagabond) [inline src/ledger/v1/blockchain_ledger_v1.erl:484]: Can we pull it from the header now?

Comment (evanmcc) [reply src/ledger/v1/blockchain_ledger_v1.erl:484]: yes, I meant to do that.

File: src/ledger/v1/blockchain_ledger_snapshot_v1.erl (hunk 1 above)
Comment (evanmcc) [reply src/ledger/v1/blockchain_ledger_snapshot_v1.erl:301]: see #462

**Comment by Vagabond** [comment]
I know dialyzer is a trainwreck right now, but there's some possibly relevant dialyzer warnings from this change

**Comment by Vagabond** [comment]
```
%%% blockchain_simple_SUITE ==> {'EXIT',
    {shutdown,
        {failed_to_start_child,blockchain_worker,
            {function_clause,
                [{blockchain,'-mark_upgrades/2-fun-0-',
                     [<<"gateway_v2">>],
                     [{file,"/home/andrew/blockchain-core/src/blockchain.erl"},
                      {line,183}]},
                 {lists,foreach,2,[{file,"lists.erl"},{line,1338}]},
                 {blockchain,mark_upgrades,2,
                     [{file,"/home/andrew/blockchain-core/src/blockchain.erl"},
                      {line,183}]},
                 {blockchain,new,4,
                     [{file,"/home/andrew/blockchain-core/src/blockchain.erl"},
                      {line,153}]},
                 {blockchain_worker,load_chain,3,
                     [{file,
                          "/home/andrew/blockchain-core/src/blockchain_worker.erl"},
                      {line,874}]},
                 {blockchain_worker,init,1,
                     [{file,
                          "/home/andrew/blockchain-core/src/blockchain_worker.erl"},
                      {line,312}]},
                 {gen_server,init_it,2,[{file,"gen_server.erl"},{line,374}]},
                 {gen_server,init_it,6,
                     [{file,"gen_server.erl"},{line,342}]}]}}}}```

**Pull request merged**